twake/frontend/src/app/components/AutoComplete/AutoComplete.tsx

```diff
@@ -1,141 +1,175 @@
 import React, { Component } from 'react';
-
+import AutoHeight from '../AutoHeight/AutoHeight';
+import Input from '../Inputs/Input';
 import './AutoComplete.scss';
-import AutoHeight from 'components/AutoHeight/AutoHeight.js';
-import Input from 'components/Inputs/Input.js';
 
-export default class AutoComplete extends Component {
-  /*
-        props : {
-            placeholder : string,
-            search : Array of search function,
-            max : Array of maximum of proposition
-            onSelect : callback on selection
-            renderItem(item) : how to render item
-            renderItemChoosen(item) : how to add item in input
-            regexHooked : on which regex start auto completed ex : /\@([A-z])/ -> check match on [A-z] after @
-            onChange
-            value
-            onChangeCurrentList : return currentList,selected when update of selected proposition or proposition
-            hideResult
-            onBackspace : when backspace on empty input
-            disableNavigationKey
-            showResultsOnInit
-            position: top / bottom
-        }
-    */
+type State = {
+  currentList: any[];
+  selected: number;
+  resultPosition: string;
+  selectedUser: object;
+  focused: any;
+};
+
+type Props = {
+  onResize?: any;
+  autoHeight?: any;
+  className?: string;
+  placeholder?: string;
+  search: ((text: string, cb: (arr: any) => any) => void)[];
+  max: number[];
+  onSelect?: (obj: object, other: any) => void;
+  renderItem: ((obj: object) => any)[];
+  renderItemChoosen: ((el: any) => any)[];
+  regexHooked: any;
+  onChange?: any;
+  value?: any;
+  onChangeCurrentList?: any;
+  hideResult?: any;
+  onBackspace?: any;
+  disableNavigationKey?: any;
+  showResultsOnInit?: any;
+  position?: any;
+  onHide?: any;
+  onEscape?: any;
+  keyUp?: any;
+  onKeyUp?: any;
+  onKeyDown?: any;
+  onKeyPress?: any;
+  onFocusChange?: any;
+  big?: boolean;
+  small?: boolean;
+  autoFocus?: any;
+};
 
-  constructor(props) {
+export default class AutoComplete extends Component<Props, State> {
+  currentIdFromList: string = '';
+  is_open: boolean = false;
+  input: any;
+  container: any;
+  outsideClickListener: any;
+  currentRegexUsed: any = -1;
+
+  constructor(props: any) {
     super(props);
-    this.props = props;
+
     this.state = {
       currentList: [],
-      currentRegexUsed: -1,
       selected: -1,
       resultPosition: '',
+      selectedUser: {},
+      focused: '',
     };
 
     this.keyUp = this.keyUp.bind(this);
     this.keyDown = this.keyDown.bind(this);
 
-    var that = this;
-    window.addEventListener('keydown', function (evt) {
```

Comment:
Need to be changed to also remove the event listener on unmount

Reply:
Done :+1: